cluster-autoscaler/core/static_autoscaler.go

```diff
@@ -168,6 +169,8 @@ func NewStaticAutoscaler(
 	ndt := deletiontracker.NewNodeDeletionTracker(0 * time.Second)
 	scaleDown := legacy.NewScaleDown(autoscalingContext, processors, ndt, deleteOptions)
 	actuator := actuation.NewActuator(autoscalingContext, clusterStateRegistry, ndt, deleteOptions)
+	autoscalingContext.ScaleDownActuator = actuator
```

Comment:
It's a bit weird to have the field in `AutoscalingContext`, but populate it when creating `StaticAutoscaler`. Would it be possible to untangle the dependencies somehow? I'm not sure how feasible that is without a bigger refactor though.

Reply:
I don't think it is without a large refactor. To do this we would need to create `Actuator` before creating `AutoscalingContext`. `Actuator` would need to get passed `CloudProvider`, `AutoscalingOptions`, `AutoscalingKubeClients`, `AutoscalingOptions` and `ClusterSnapshot` unpacked from the `AutoscalingContext` wrapper. This looks like a lot of refactor and IMHO would actually make this less readable than it is now.

`AutoscalingContext` is also initialised in this method, so it is not that far from the definition. Also we are already passing `ScaleDownWrapper` in a similar fashion two lines below - it would be nice to refactor both, which require even more changes.

Reply:
Agreed it's not good now, but at least the confusion is mostly scoped to `NewStaticAutoscaler`. Could you add a TODO for the bigger refactor somewhere in there? And/or somewhere around `AutoscalingContext` definition? Maybe the next person adding something could pick the refactor up.